BJS2-3646 - Endpoints для контроллеров - Unicorn BC3

hhzktoeto opened the pull request.

hhzktoeto committed:
Изменил валидацию DTO

теперь поля DTO валидируются с помощью аннотаций
удалил ненужные методы, в тестах тоже
убрал ненужные импорты

hhzktoeto committed:
Добавил эндпоинты + поправки

добавил эндпоинты в VacancyController
Заменил валидацию @NotBlank на @NotNull для всех полей, содержащих Long

IrinaVereshchagina13 commented on src/main/java/faang/school/projectservice/controller/vacancy/VacancyController.java:
```diff
@@ -3,37 +3,55 @@
+    @DeleteMapping
+    public void delete(@RequestBody VacancyDto vacancyDto) {
         vacancyService.delete(vacancyDto);
     }
```
Удалять будет проще по id

hhzktoeto committed:
Исправления после review

метод delete теперь принимает только vacancyId
исправил логику выполнения метода в сервисе
исправил тесты
